File: LearningProject/Models/Question.cs
using System.ComponentModel.DataAnnotations;

namespace LearningProject.Models
{
    public class Question
    {
        public int Id { get; set; }
        [Range(1, 100, ErrorMessage = "Marks must be between 1 and 100")]
        public int Marks { get; set; }
        public string Level { get; set; }
        public int? ActivityId { get; set; }

        public bool IsPublic { get; set; }
        public int QuestionTypeId { get; set; }

        // Navigation properties
        public QuestionType QuestionType { get; set; }

        // Navigation property for QuestionExam
        public ICollection<Answer>? Answers { get; set; }
    }
}

File: client/src/app/core/models/Question.ts
import { Answer } from "./Answer";
import { ExamQuestion } from "./ExamQuestion";
import { QuestionType } from "./QuestionType";

export interface Question {
    id: number;
    marks: number;
    name?: string | null;
    activityId?: number | null;
    isPublic: boolean;
    questionTypeId: number;
    questionType?: QuestionType | null;
    answers?: Answer[] | null;
    examQuestions?: ExamQuestion[] | null;
  }
  